docs/dev/analysis/tbl-cell-borders.rst

```diff
@@ -0,0 +1,92 @@
+
+Table Cell Borders
+------------------
+
+In PowerPoint, a table cell can have borders on all 4 sides, plus two diagonal lines (bottom left to top right and
+top left to bottom right) for a total of 6 possible borders, each specified by a child element of the table cell
+properties element.
+
+Each border is further customized by a set of attributes and child elements common to all borders (`LineFormat`).
+
+In order to provide `LineFormat` functionality, each border `_ln` object is wrapped in a `_TableBorderLine`,
+which implements the `get_or_add_ln` method, and can be then wrapped by `LineFormat`.
+
+Protocol
+--------
+
+**Starting with a default cell** A new cell has no borders or style.
+New borders can be created by accessing the borders property::
+
+    >>> table = shapes.add_table(rows, columns, left, top, width, height).table
+    >>> cell = table.rows[0].cells[0]
+    >>> cell
+    <pptx.shapes.table._Cell object at 0x...>
+    >>> cell.borders
+    <pptx.shapes.table._TableBorders object at 0x...>
+
+**Create a border** A cell border is created by applying line width and color for its `LineFormat`.
+This can be done with any one of the 6 cell borders::
+
+    >>> cell.borders.top.width = Pt(1)
+    >>> cell.borders.top.color.rgb = RGBColor(0, 0, 0)
+    >>> cell.borders.top._ln.xml
+    <a:lnT w="12700">
+      <a:solidFill>
+        <a:srgbClr val="000000"/>
+      </a:solidFill>
+    </a:lnT>
+    >>> cell._tc.xml
+    <a:tc>
+      <a:txBody>
+        <a:bodyPr/>
+        <a:lstStyle/>
+        <a:p/>
+      </a:txBody>
+      <a:tcPr>
+        <a:lnT w="12700">
+          <a:solidFill>
+            <a:srgbClr val="000000"/>
+              </a:solidFill>
+        </a:lnT>
+      </a:tcPr>
+    </a:tc>
+
+**Apply different line styles** Cell border work with `LineFormat`, so any other line style can also be applied::
+
+    >>> cell.borders.top.dash_style = MSO_LINE_DASH_STYLE.DASH_DOT_DOT
+    >>> cell.borders.top._ln.xml
+    <a:lnT w="12700">
+      <a:solidFill>
+        <a:srgbClr val="000000"/>
+      </a:solidFill>
+      <a:prstDash val="lgDashDotDot"/>
+    </a:lnT>
+
+
```

Comment:
We should probably insert a brief 'Scope' section below the protocol so it's clear what you plan to implement. I'm thinking it's something like:

```rst
* `_Cell.borders` -> `_CellBorders`
* `_Cell.borders = None` clears all cell borders?
* `_CellBorders.top` -> `_CellBorderLine`
* `_CellBorders.top = None` clears any defined border?
* ... other borders (were you planning on diagonals as well?)
* `_CellBorderLine` inherits `LineFormat` (or whatever the plan is there)
```

I'm not sure how you were thinking to get rid of borders, but that's probably something we need to consider. Off-hand I'm thinking the two assignments of `None` above are not bad API-wise, unless there's an existing way to do that for `LineFormat` objects we should stay consistent with.